app/submitter/convert_payload_0_0_1.py

```diff
@@ -126,9 +126,8 @@ def _get_checkbox_answer_data(
         if option:
             if "detail_answer" in option:
                 detail_answer = answer_store.get_answer(option["detail_answer"]["id"])
-                # if the user has selected an option with a detail answer we need to find the detail answer value it refers to.
-                # the detail answer value can be empty, in this case we just use the main value (e.g. other)
-                user_answer = detail_answer.value or user_answer  # type: ignore
+                if detail_answer:
+                    user_answer = detail_answer.value  # type: ignore
```

Comment:
I would add a comment that the ignore is because answer type can be any non strings, but this expects to be a string.

Reply:
Added a type hint comment